classes/war/WarLogDto.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+class WarLogDto {
+    public int $log_id;
+    public string $log_type;
+    public ?int $user_id = null;
+    public ?string $user_name = null;
+    public int $village_id;
+    public ?string $village_name = null;
+    public ?int $relation_id;
+    public int $rank = 0;
+    public int $infiltrate_count = 0;
+    public int $reinforce_count = 0;
+    public int $raid_count = 0;
+    public int $loot_count = 0;
+    public int $damage_dealt = 0;
+    public int $damage_healed = 0;
+    public int $defense_gained = 0;
+    public int $defense_reduced = 0;
+    public int $resources_stolen = 0;
+    public int $resources_claimed = 0;
+    public int $patrols_defeated = 0;
+    public int $regions_captured = 0;
+    public int $villages_captured = 0;
+    public int $pvp_wins = 0;
+    public int $points_gained = 0;
+    public int $war_score = 0;
+    public int $objective_score = 0;
+    public int $resource_score = 0;
+    public int $battle_score = 0;
+
+    public function __construct(array $row, string $log_type) {
+        foreach ($row as $key => $value) {
+            $this->$key = $value;
+        }
```

Comment:
I know it's more boilerplate but I think we should avoid this pattern wherever possible, and write out the lines to set each value from the array individually.

If we have this magic that auto-sets the values it can hide issues where a new DB column is added but isn't in code, or there's a typo on a column, etc

Could structure this similar to a pattern used elsewhere:

```php
class WarLogDto {
  public function __construct(
    public int $log_id,
    public string $log_type,
    // ... 
  ) {}

  public static function fromDbArray(array $db_row, string $log_type): WarLogDto {
    return new WarLogDto(
      log_id: $db_row['log_id'],
      log_type: $log_type,
    );
  }
}
```